File: client/src/main/scala/stasis/client/encryption/secrets/DeviceMetadataSecret.scala
package stasis.client.encryption.secrets

import org.apache.pekko.util.ByteString
import stasis.client.encryption.Aes
import stasis.client.encryption.stream.CipherStage

final case class DeviceMetadataSecret(
  iv: ByteString,
  private val key: ByteString
) extends Secret {
  def encryption: CipherStage = Aes.encryption(key, iv)
  def decryption: CipherStage = Aes.decryption(key, iv)
}

object DeviceMetadataSecret {
  def apply(iv: ByteString, key: ByteString): DeviceMetadataSecret =
    new DeviceMetadataSecret(iv, key)
}

File: client-android/lib/src/main/kotlin/stasis/client_android/lib/encryption/secrets/DeviceMetadataSecret.kt
package stasis.client_android.lib.encryption.secrets

import okio.ByteString
import okio.Source
import stasis.client_android.lib.encryption.Aes

data class DeviceMetadataSecret(
    val iv: ByteString,
    private val key: ByteString
) : Secret() {
    fun encryption(source: Source): Source = Aes.encryption(source, key, iv)
    fun decryption(source: Source): Source = Aes.decryption(source, key, iv)
}
